### DISCUSSION TITLE
Language is not taking the correct one in production

### DISCUSSION THREAD

**Discussion by vmmansilla93** [stephanrauh/ngx-extended-pdf-viewer#2346] [General]
Hello @stephanrauh,
I have a problem with the language of the ngx-extended-pdf-viewer, I have setted it in `nl` with the property of `[language]="'nl'"` and in local is working:

![imagen_2024-05-29_103343876](https://github.com/stephanrauh/ngx-extended-pdf-viewer/assets/164879765/a4a1e1a2-b98c-4cf1-a6ec-0dcd57d6d5af)

And this is the code:

```html
<ngx-extended-pdf-viewer #pdfViewer
  [language]="'nl'" [hidden]="isLoading"class="pdf-viewer" theme="dark"
  backgroundColor="var(--surface-100)" [textLayer]="true" [showDrawEditor]="false"
  [showHighlightEditor]="false" [showOpenFileButton]="false" [showTextEditor]="false"
  [showStampEditor]="false" [showFindButton]="false" [showSecondaryToolbarButton]="false">
</ngx-extended-pdf-viewer>
```

But when it is deployed to production with the same code it's not taking anymore nl language, it's using english instead of nl:

![image](https://github.com/stephanrauh/ngx-extended-pdf-viewer/assets/164879765/09d7c281-14ae-4d0f-a84e-14e2f85378d5)

Any help to solve this or what could be my error?
Many thanks in advance!

**Comment by stephanrauh** [comment]
Maybe the language files are missing in production. Are there any error messages in the network tab or the console?

Here's a working example - as things go, with a similar language (nl-be): https://pdfviewer.net/extended-pdf-viewer/i18n

**Comment by vmmansilla93** [comment]
Mmhh maybe is this the problem, I can see that in both environments the documents are being donwloaded, but in production they are weird:
Local:
![image](https://github.com/stephanrauh/ngx-extended-pdf-viewer/assets/164879765/46a4ab7a-d5e2-480f-b991-f1fcbef95160)

Prod:
![image](https://github.com/stephanrauh/ngx-extended-pdf-viewer/assets/164879765/73a32c43-088f-44f4-89a2-e7ad16637817)

But both english and nl in prod are white like this. What I didn't do is to copy the nl.locale asset in my project as the demo you said, I'll try this approach!

**Reply by vmmansilla93** [reply]
I tried it but still the same, don't know what can be, I even added the same reload for change the language as the demo but doesn't change any language in production, doesn't load the file aswell, only blank files, even the english one. In local everything works perfectly...

**Reply by stephanrauh** [reply]
Probably the files aren't empty. It's more likely that Angular loads the `index.html`. That's the default when it hits a file that doesn't exist - it allows for routing. You can see that if you open the "reponse" tab instead of the "preview" tab.

The folder `assets/locale` also has to contain the `locale.json` file. But according to your screenshot, that file seems to be there.
<img width="991" alt="image" src="https://github.com/stephanrauh/ngx-extended-pdf-viewer/assets/3045767/00588773-69ab-4fa8-91a6-1884368f8e22">

**Comment by SoyDiego** [comment]
Hi @stephanrauh, thanks for your reply.
We were trying your advice. The `.ftl` files are blocked by Organization Security Policies and we realized `.json` and `.txt`extensions are allowed.
How could we change `.ftl` extension to  `.json` or another workaround to fix it. If you could guide us or give an idea will help and we can try to do a PR if possible. 

Thanks for your time!

**Reply by stephanrauh** [reply]
What? That's crazy! I've been working in a major company myself, so I know these things happen, but blocking *.ftl files sounds just weird.

As for the workaround: As far as I remember, the pdf.js team used a standard library, but I suspect they've put the file into the project for some reason: https://github.com/stephanrauh/pdf.js/blob/bleeding-edge/external/importL10n/locales.mjs

If that's true, you can modify this file and send me a pull request. But I don't think it's simply a matter of renaming the files. You know, every new version of pdf.js modifies the *.ftl files, and I want new versions of ngx-extended-pdf-viewer to benefit from these changes. If we rename the file, I'll end up in merge hell.

But maybe this approach works:
- You find out how to configure the `angular.json` such that is renames the *.ftl files to *.json. Oh, and please tell me about it, so I can document it in the showcase.
- Once you've managed that, please send me a PR that reads *.ftl files if they're available, and defaults to loading *.json files otherwise.

I hope I didn't forget anything. In any case, let me quote what I've often read on Spanish road signs: ¡disculpen las molestias!
